pkg/nvmeclient/nvme_client.go

```diff
@@ -521,11 +543,14 @@ func ConnectAllNVMEDevices(logPageEntries []*hostapi.NvmeDiscPageEntry, hostnqn
 				if perr.Status == CONN_ALREADY_CONNECTED {
 					continue
 				} else {
-					// This warn will occur every 5 sec in case the node is down.
-					// discovery service will still report this controller to connect to but we will fail to connect.
-					// we can't deduce that if the DS is down on that node we will fail to connect cause there might be a network partition
-					// on the discovery-service or the DS is down on that node but the IO controller is still accessible.
-					logrus.WithError(perr).Warnf("failed to connect IO controller. In case we have a node down, discovery provides log-page with it's connect info. we will probably fail to connect to it.")
+					ctrlID, err = connectNVMEDevicesWithRetry(request)
+					if errors.As(err, &perr) {
+						// This warn will occur every 5 sec in case the node is down.
+						// discovery service will still report this controller to connect to but we will fail to connect.
+						// we can't deduce that if the DS is down on that node we will fail to connect cause there might be a network partition
+						// on the discovery-service or the DS is down on that node but the IO controller is still accessible.
+						logrus.WithError(perr).Warnf("failed to connect IO controller. In case we have a node down, discovery provides log-page with it's connect info. we will probably fail to connect to it.")
```

Comment:
Please change the warning to be proper English. Something like this:
```
"failed to connect to IO controller. This may be a transient error or due to a node down. Continuing to attempt connection until the discovery-service stops providing the node's address."
```

Reply:
@muliby-lb sure. did you mean "**starts** providing the node's address"?

Reply:
I meant stops providing the old node's address (the one that is down) but you can also think of it as starts providing the new node's address. To clarify, you can change it to "... stops providing the **down** node's address".